readthedocs/projects/tasks.py

```diff
@@ -1430,25 +1430,35 @@ def _manage_imported_files(version, path, commit):
             else:
                 model_class = ImportedFile
 
-            dirpath = os.path.join(
-                root.replace(path, '').lstrip('/'), filename.lstrip('/')
-            )
             full_path = os.path.join(root, filename)
+            relpath = os.path.relpath(full_path, path)
             md5 = hashlib.md5(open(full_path, 'rb').read()).hexdigest()
+            obj_attributes = dict(
+                project=version.project,
+                version=version,
+                path=relpath,
+                name=filename,
+            )
             try:
-                # pylint: disable=unpacking-non-sequence
                 obj, created = model_class.objects.get_or_create(
-                    project=version.project,
-                    version=version,
-                    path=dirpath,
-                    name=filename,
+                    **obj_attributes,
                 )
             except model_class.MultipleObjectsReturned:
-                log.warning('Error creating ImportedFile')
-                continue
+                log.warning('Fixing duplicated objects')
+                duplicateds = (
+                    model_class.objects
+                    .filter(**obj_attributes)
+                    .order_by('-modified_date')[1:]
```

Comment:
I'm removing all duplicated objects but the last one

Reply:
Seems like we shouldn't need to do this. We should do it in a migration, and write the code so it doesn't happen, no?

Reply:
Yeah, that is better. But I'll need to search where we are duplicating these objects (if this is still happening), I think there was a PR that changed this, so maybe they are residuals? But I faced this locally somehow, need to investigate more.